pandas/tests/test_algos.py

```diff
@@ -570,6 +570,13 @@ def test_object_refcount_bug(self):
         for i in range(1000):
             len(algos.unique(lst))
 
+    def test_index_returned(self, index):
+        # GH#57043
+        index = index.repeat(2)
+        result = algos.unique(index)
+        expected = index.unique()
```

Comment:
Is it doable to specify `expected` explicitly here instead of using `index.unique()`?

Reply:
I think we can adapt the test from https://github.com/pandas-dev/pandas/blob/4692686a750b95884e1d5a4f1313614c71c57213/pandas/tests/base/test_unique.py#L12